contracts/user/UserManager.sol

```diff
@@ -808,27 +814,30 @@ contract UserManager is Controller, IUserManager, ReentrancyGuardUpgradeable {
     }
 
     function withdrawRewards() external whenNotPaused nonReentrant {
+        // slither-disable-next-line unused-return
         comptroller.withdrawRewards(msg.sender, stakingToken);
     }
 
     /**
-     *  @dev Repay user's loan overdue, called only from the lending market
+     *  @dev Repay user's loan overdue, only called by UToken
      *  @param account User address
      *  @param token The asset token repaying to
      *  @param lastRepay Last repay block number
+     *  @return counter Number of frozen stakers, so we know if repay overdue loan works.
      */
     function repayLoanOverdue(
         address account,
         address token,
         uint256 lastRepay
-    ) external override whenNotPaused onlyMarketOrAdmin {
+    ) external override whenNotPaused onlyMarketOrAdmin returns (uint8 counter) {
         address[] memory stakerAddresses = getStakerAddresses(account);
-        uint256 addressesLength;
```

Comment:
Does this change need to be onchain?

Reply:
This is an optimization to cache the number of addresses used in the loop.